src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -135,7 +135,8 @@ static MsQuicApi()
                 return;
             }
 
-            if (NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out IntPtr msQuicHandle))
+            if (NativeLibrary.TryLoad(Interop.Libraries.MsQuic, typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out IntPtr msQuicHandle) ||
+                NativeLibrary.TryLoad($"{Interop.Libraries.MsQuic}.{MsQuicVersion}", typeof(MsQuicApi).Assembly, DllImportSearchPath.AssemblyDirectory, out msQuicHandle))
```

Comment:
This new line would be Linux specific, right?

Reply:
and macOS. On Windows the first load _should_ always work (at least for now when we include it in run time)
I _could_ make the fallback platform specific.

Reply:
I don't think it's necessary. Probably. But actually another thought: Does this pose any kind of security problem? On Linux, once we move to the `libmsquic.1` model and `libmsquic` isn't there, what if someone uses this logic to drop in their own `libmsquic`? Do we care?

Reply:
That goes to library search path. If you can make system to search private locations, you can already put in untrusted code. (or do tricks with LD_PRELOAD) 

if/when we abandon dotnet/msquic we can probably remove the fallback. We _may_ create issue to track it.

Reply:
Sounds good. Thanks!

Reply:
BTW we do something similar of OpenSSL and even worse for libicu:
https://github.com/dotnet/runtime/blob/86f4a57b5db4cda494a5c1ce29863624713a92c0/src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c#L323-L337

This is the type probing I was hoping to avoid.